src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs

```diff
@@ -241,5 +244,10 @@ private string ValidateSolutionPath(string basePath, string solutionPath)
 
             return solutionPath;
         }
+
+        public void UpdateBasePath(string newBaseDirectory)
```

Comment:
Do we have to update the `BasePath` later than the creation of the options? Why not set the `BasePath` with a value that is passed. If it was not passed set it to the current directory (should  be the test project directory). That way the config doesn't have to be updated later. And validation can just be done like all other configs.

Reply:
That's a great idea. I was considering it a bit, but it was too late(in the night) to consider the scope of use of the base path.

I thought the Options would be immutable, i would also consider it a value object, but then i noticed the TestRunner being get set as well.

I will change this to set the basepath to the provided test project path if one is provided.

Reply:
You are right, the `TestRunner` is a bit of a flaw in the design. It is set later to VsTest if we detect some values in the project file. It would be better to not change the option there but the actual testrunner object that is used. For now this was the easy solution.

Reply:
@richardwerkman If we only change the actual object then the used configuration will differ from what the user chose in an implicit way. This means in other parts of stryker we will not be able to determine what testrunner is used (as we only have the options object in most places). That would not be ideal and could lead to bugs.